protocol/contracts/profit/src/state/mod.rs

```diff
@@ -129,27 +107,23 @@ impl ConfigManagement for State {
 }
 
 impl State {
-    pub fn new(config: Config) -> Self {
-        Self(StateEnum::OpenTransferChannel(OpenTransferChannel::new(
-            config,
-        )))
+    pub(crate) fn start(config: Config, dex: ConnectionParams) -> (Self, Batch) {
+        let init_state = IcaConnector::new(OpenIca::new(config, dex));
+
+        let response = init_state.enter();
+        let state: State = init_state.into();
+        (state, response)
     }
 
-    pub fn load(storage: &dyn Storage) -> ContractResult<Self> {
+    pub(crate) fn load(storage: &dyn Storage) -> ContractResult<Self> {
         STATE.load(storage).map_err(Into::into)
     }
 
-    pub fn store(&self, storage: &mut dyn Storage) -> ContractResult<()> {
+    pub(crate) fn store(&self, storage: &mut dyn Storage) -> ContractResult<()> {
```

Comment:
The visibility of this fn is limited to the visibility of the `struct` it is defined in. Therefore no need to specialize its visibility. Please remove.